db/clientlist_virtual_reader.cc

```diff
@@ -0,0 +1,237 @@
+        if (_val.size() < rhs.size()) {
+            _val.swap(rhs);
+        }
+        _val.reserve(_val.size() + rhs.size());
+        _val.insert(_val.end(), std::make_move_iterator(rhs.begin()), std::make_move_iterator(rhs.end()));
+        return make_ready_future<>();
+    }
+
+    std::vector<system_keyspace::client_data> get() && {
+        return std::move(_val);
+    }
+};
+
+class ck_comparator {
+    schema_ptr _schema;
+public:
+    using port_type = decltype(system_keyspace::client_data::port);
+
+    ck_comparator(schema_ptr schema) : _schema(schema) {}
+
+    int operator()(const clustering_key_prefix& lhs, const clustering_key_prefix& rhs) {
+        return clustering_key_prefix::prefix_equal_tri_compare(*_schema)(lhs, rhs);
+    }
+    int operator()(const port_type& lhs, const clustering_key_prefix& rhs) {
+        const auto lhs_bytes = data_type_for<port_type>()->decompose(lhs);
+        return operator()(clustering_key_prefix::from_single_value(*_schema, lhs_bytes), rhs);
+    }
+    int operator()(const clustering_key_prefix& lhs, const port_type& rhs) {
+        const auto rhs_bytes = data_type_for<port_type>()->decompose(rhs);
+        return operator()(lhs, clustering_key_prefix::from_single_value(*_schema, rhs_bytes));
+    }
+};
+}  // anon. namespace
+
+static future<std::vector<system_keyspace::client_data>> get_all_clients_data(
+        distributed<cql_transport::cql_server>& distributed_cql_server) {
+    return distributed_cql_server.map_reduce(clients_reducer{},
+            [] (const cql_transport::cql_server& srv) {
+        const auto& conn_list = srv.get_connections_list();
+        return do_with(std::vector<system_keyspace::client_data>{},
+                [&conn_list] (auto& per_shard_result) {
+            return do_for_each(conn_list.begin(), conn_list.end(),
+                    [&per_shard_result] (const auto& conn) {
+                system_keyspace::client_data cd;
+                cd.ip = conn.get_client_state().get_client_address().addr();
+                cd.port = conn.get_client_state().get_client_port();
+                cd.connection_stage = conn.get_client_state().get_auth_state();
+                cd.protocol_version = conn.get_protocol_version();
+                cd.username = conn.get_client_state().user()->name;
+
+                per_shard_result.emplace_back(std::move(cd));
+            }).then([psr = std::move(per_shard_result)] {
+                return make_ready_future<std::vector<system_keyspace::client_data>>(std::move(psr));
+            });
+        });
+    });
+}
+
+// Clusters clients by IP and sorts them by port number within clusters.
+future<std::vector<system_keyspace::client_data>>
+clientlist_mutation_reader::filter_and_sort_by_port(std::vector<system_keyspace::client_data>&& clients) const {
+    std::vector<system_keyspace::client_data> sorted_clients;
+    sorted_clients.reserve(clients.size());
+    
+    std::unordered_map<
```

Comment:
So you're defining an `unordered_map`, then copying data from it into a `vector` and sorting the `vector`.
Hmmmm...
https://en.cppreference.com/w/cpp/container/map